codalab/apps/web/tasks.py

```diff
@@ -876,27 +880,88 @@ def send_chahub_updates():
 
 
 @task(queue='site-worker')
-def create_storage_statistic_datapoint():
-        total_bytes = storage_get_total_use(BundleStorage)
-        data = {
-            'total_use': total_bytes,
-            'bucket_name': BundleStorage.bucket.name
+def create_storage_analytics_snapshot():
+    logger.info("Task create_storage_analytics_snapshot started")
+    t = time.process_time()
+
+    # Retrieve the last storage usage history point
+    bucket = BundleStorage.bucket
+    last_storage_usage_history_point = StorageUsageHistory.objects.filter(bucket_name=bucket.name).order_by('-at_date').first()
+
+    current_datetime = datetime.datetime.now(datetime.timezone.utc)
+    last_storage_usage_history_date = last_storage_usage_history_point.at_date if last_storage_usage_history_point else current_datetime - datetime.timedelta(days=1000)
+
+    # Prepare the storage usage history points and total usage
+    days = range(1, int((current_datetime - last_storage_usage_history_date).days) + 1)
```

Comment:
Pulling `(current_datetime - last_storage_usage_history_date).days)` to a separate variable may improve readability.

IE: 
```
days_count = (current_datetime - last_storage_usage_history_date).days)
days = range(1, days + 1)
```

Reply:
done